src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp

```diff
@@ -1026,58 +1029,111 @@ template <typename ParallelComponent, typename DbTagsList>
-    const auto direction_from_neighbor = orientation(direction.opposite());
+  if constexpr (detail::has_boundary_correction_v<system>) {
+    const auto& time_step_id = db::get<::Tags::TimeStepId>(box);
+    const auto& all_mortar_data =
+        db::get<evolution::dg::Tags::MortarData<volume_dim>>(box);
+
+    for (const auto& [direction, neighbors] : element.neighbors()) {
+      const auto& orientation = neighbors.orientation();
+      const auto direction_from_neighbor = orientation(direction.opposite());
+
+      for (const auto& neighbor : neighbors) {
+        const std::pair mortar_id{direction, neighbor};
+
+        std::pair<Mesh<volume_dim - 1>, std::vector<double>>
+            remote_boundary_data_on_mortar =
+                *all_mortar_data.at(mortar_id).local_mortar_data();
+        ASSERT(time_step_id == all_mortar_data.at(mortar_id).time_step_id(),
+               "The current time step id of the volume is "
+                   << time_step_id
+                   << "but the time step id on the mortar with mortar id "
+                   << mortar_id << " is "
+                   << all_mortar_data.at(mortar_id).time_step_id());
+
+        // Reorient the data to the neighbor orientation if necessary
+        if (not orientation.is_aligned()) {
+          ERROR("Currently don't support unaligned meshes");
+          // remote_boundary_data_on_mortar.orient_on_slice(
+          //     mortar_meshes.at(mortar_id).extents(), direction.dimension(),
+          //     orientation);
+        }
 
-    for (const auto& neighbor : neighbors) {
-      const ::dg::MortarId<volume_dim> mortar_id{direction, neighbor};
-
-      // Make a copy of the local boundary data on the mortar to send to the
-      // neighbor
-      ASSERT(all_mortar_data.find(mortar_id) != all_mortar_data.end(),
-             "Mortar data on mortar "
-                 << mortar_id
-                 << " not available for sending. Did you forget to collect "
-                    "the data on mortars?");
-      auto mortar_data_sent_to_neighbor =
-          all_mortar_data.at(mortar_id).local_data(temporal_id);
-
-      // Reorient the data to the neighbor orientation if necessary
-      if (not orientation.is_aligned()) {
-        mortar_data_sent_to_neighbor.orient_on_slice(
-            mortar_meshes.at(mortar_id).extents(), dimension, orientation);
+        std::tuple<Mesh<volume_dim - 1>, std::optional<std::vector<double>>,
+                   std::optional<std::vector<double>>, ::TimeStepId>
+            data{remote_boundary_data_on_mortar.first,
+                 {},
+                 {std::move(remote_boundary_data_on_mortar.second)},
+                 db::get<tmpl::conditional_t<Metavariables::local_time_stepping,
+                                             ::Tags::Next<::Tags::TimeStepId>,
+                                             ::Tags::TimeStepId>>(box)};
+
+        // Send remote data to neighbor
```

Comment:
It's not at all clear to me what this means, though perhaps it's related to my other comment. Does "remote" mean "neighbor" here, as it did in a previous PR? Please clarify...